handler/src/main/java/io/netty/handler/ssl/JdkSslClientContext.java

```diff
@@ -244,24 +254,26 @@ public JdkSslClientContext(File trustCertCollectionFile, TrustManagerFactory tru
             File keyCertChainFile, File keyFile, String keyPassword, KeyManagerFactory keyManagerFactory,
             Iterable<String> ciphers, CipherSuiteFilter cipherFilter, JdkApplicationProtocolNegotiator apn,
             long sessionCacheSize, long sessionTimeout) throws SSLException {
-        super(newSSLContext(toX509CertificatesInternal(
+        super(newSSLContext(null, toX509CertificatesInternal(
                 trustCertCollectionFile), trustManagerFactory,
                 toX509CertificatesInternal(keyCertChainFile), toPrivateKeyInternal(keyFile, keyPassword),
                 keyPassword, keyManagerFactory, sessionCacheSize, sessionTimeout), true,
                 ciphers, cipherFilter, apn, ClientAuth.NONE, null, false);
     }
 
-    JdkSslClientContext(X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
+    JdkSslClientContext(Provider sslContextProvider,
+                        X509Certificate[] trustCertCollection, TrustManagerFactory trustManagerFactory,
                         X509Certificate[] keyCertChain, PrivateKey key, String keyPassword,
                         KeyManagerFactory keyManagerFactory, Iterable<String> ciphers, CipherSuiteFilter cipherFilter,
                         ApplicationProtocolConfig apn, String[] protocols, long sessionCacheSize, long sessionTimeout)
             throws SSLException {
-        super(newSSLContext(trustCertCollection, trustManagerFactory, keyCertChain, key, keyPassword,
-                keyManagerFactory, sessionCacheSize, sessionTimeout), true,
-                ciphers, cipherFilter, toNegotiator(apn, false), ClientAuth.NONE, protocols, false);
+        super(newSSLContext(sslContextProvider, trustCertCollection, trustManagerFactory,
+                            keyCertChain, key, keyPassword, keyManagerFactory, sessionCacheSize, sessionTimeout),
+                true, ciphers, cipherFilter, toNegotiator(apn, false), ClientAuth.NONE, protocols, false);
     }
 
-    private static SSLContext newSSLContext(X509Certificate[] trustCertCollection,
+    private static SSLContext newSSLContext(Provider sslContextProvider,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00107)